app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Generator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._size = 8
+        self._threshold = 0.3
+        self._table: list[list] = [[] for _ in range(self._size)]
+        self._resize_const = 2
+
+    def print_table(self) -> None:
+        print(self._table)
+
+    def __getitem__(self, key: any) -> all:
+        hash_key = hash(key)
+        table_index = hash_key % self._size
+
+        for _ in range(self._size):
+            if self._table[table_index] and self._table[table_index][1] == key:
+                return self._table[table_index][0]
+            else:
+                table_index = next(self._gen_next_slot(table_index))
+                if (self._table[table_index]
+                        and self._table[table_index][1] == key):
+                    return self._table[table_index][0]
+
+        raise KeyError
```

Comment:
The `KeyError` should include a message indicating which key was not found. Consider raising `KeyError(f'Key {key} not found')` to provide more informative error messages.